core/flyout_base.ts

```diff
@@ -996,14 +1034,17 @@ export abstract class Flyout extends DeleteArea implements IFlyout {
 
   /**
    * Create and place a rectangle corresponding to the given block.
+   *
    * @param block The block to associate the rect to.
    * @param x The x position of the cursor during this layout pass.
    * @param y The y position of the cursor during this layout pass.
    * @param blockHW The height and width of
    *     the block.
+   * @param blockHW.height
```

Comment:
Unfortunately, on the whole, I don't think this is something we can fully follow. The documentation-generating tool only parses the TsDoc comments, and doesn't look at the code itself. So if we want useful documentation I think we at least need to list the param name in the list. You can try it in the TsDoc playground: https://microsoft.github.io/tsdoc/ where if you remove the param entirely it will not be listed at all in the output. This tool uses the same parser that API Extractor uses.

Since internally google3 is using different tools for documentation and typechecking I think unfortunately that guidance doesn't work for us.

Right now I have the `required-param` check set to "off" but only because there is an issue with the rule in the underlying eslint plugin. These comments were added by the autofixer before I disabled that rule, but I think they should stay for that reason.

The rules as I hope they will be when the underlying bug is fixed:
- Doc comments only required on public / exported members
- param names required, but a description is not

---

In this specific case it's adding the "sub-params" that have been destructured, which probably isn't what we want and I disabled that in a later rule. So I will try to revert this specific change but in general the above comments apply.

Reply:
That all sounds very reasonable.  I've added an item to [the TS to-do list](https://github.com/google/blockly/issues/5857#issuecomment-1168877439) to update our styleguide to reflect where we differ from the Google TS styleguide.